### PR TITLE
Add Set, Set.Int, Set.String, Set.Dict docs

### LINKED ISSUE #20: Transfer Belt.Set docs
We transferred the names and type definitions, but explanations and examples are missing.

Current work: https://github.com/reason-association/reasonml.org/blob/master/pages/belt_docs/set.mdx

Belt Set docs: https://bucklescript.github.io/bucklescript/api/Belt.Set.html

Please comment on this issue if you would like to pick it up.

### PR DESCRIPTION
**Pull request by baransu**
Resolves #20

### CODE REVIEW

**Comment by nikgraf** [comment]
this is fantastic, thank you very much @baransu 👏👏👏

File: common/Util.re [hunk 1]
```diff
@@ -6,3 +6,12 @@ module ReactStuff = {
     external elementAsString: React.element => string = "%identity";
   };
 };
+
+module IntCmp =
```
Comment (nikgraf) [inline common/Util.re]: @baransu why was it necessary/useful to add IntCmp here?

Comment (baransu) [reply common/Util.re]: My little playground. Sorry 😛

Comment (baransu) [reply common/Util.re]: https://github.com/reason-association/reasonml.org/pull/35

Comment (nikgraf) [reply common/Util.re]: thx :)

**Comment by ryyppy** [comment]
Great work @baransu 🚀

**Pull request merged**